src/components/Editor/EditorInput/EditorInput.js

```diff
@@ -0,0 +1,110 @@
+import React, { createRef, useEffect, useRef, useState } from 'react'
+import { DragDropContext } from '@hello-pangea/dnd'
+import { useDispatch, useSelector } from 'react-redux'
+import { closeFile, setFocussedFileIndex, setOpenFiles } from '../EditorSlice'
+import { TabPanel, Tabs } from 'react-tabs'
+import Button from '../../Button/Button'
+import { CloseIcon } from '../../../Icons'
+import EditorPanel from '../EditorPanel/EditorPanel'
+import DraggableTab from '../DraggableTabs/DraggableTab'
+import DroppableTabList from '../DraggableTabs/DroppableTabList'
+import RunBar from '../../RunButton/RunBar'
+
+import './EditorInput.scss'
+
+const EditorInput = () => {
+  const project = useSelector((state) => state.editor.project)
+  const openFiles = useSelector((state) => state.editor.openFiles)
+  const focussedFileIndices = useSelector((state) => state.editor.focussedFileIndices)
+  const dispatch = useDispatch()
+
+  const onDragStart = (input) => {
+    const { source } = input
+    dispatch(setFocussedFileIndex({panelIndex: parseInt(source.droppableId), fileIndex: source.index}))
+  }
+
+  const onDragEnd = (result) => {
+    if(!result.destination) return
+    const { source, destination } = result
+    var openFilesData = [...openFiles]
```

Comment:
how come you're using `var` here and dotted around below? sticking to `let` and `const` is safer, unless absolutely essential (not sure I've ever had to since I started using es6) as you're risking weird scoping issues: https://www.freecodecamp.org/news/var-let-and-const-whats-the-difference/